### ISSUE TITLE
Rundeck - NoMethod Error

### ISSUE THREAD

**Issue by gangadhar01a** [sous-chefs/rundeck#232]
Hi,

I am trying to use RUNDECK supermarket cookbook to install and configure on centos 7 machine. Running chef client, local mode, resulted in undefined method error, I have included  "rundeck" as dependency in metadata.rb and I expect chef will have access to the resources/methods exist in rundeck chef cookbook.
Unfortunately, this resulted in

"NoMethodError
-------------
undefined method `server_install' for cookbook: nrhl_rundeck, recipe: test :Chef::Recipe"

I can confirm the dependency cookbooks downloaded to berks-cookbook folder.

CentOS: 7
Chef-Client: 14

**Comment by PowerSchill** [comment]
Can you provide how you are calling the server_install resource?

**Comment by gangadhar01a** [comment]
I have following recipes in recipe folder

-default.rb
#
# Cookbook:: my_cookbook
# Recipe:: default
#
# Copyright:: 2019, The Authors, All Rights Reserved.
include_recipe "#{cookbook_name}::install"

- install.rb
server_install 'Rundeck' do
  action :install
end

Ran the below command to run chef on locally, which resulted the method not found error. 
chef-client -z -o my_cookbook

**Comment by PowerSchill** [comment]
When you reference a resource from a cookbook you have to prepend the resource with the name of the cookbook so you would have something like:

```ruby
rundeck_server_install 'package' do
  action [:install]
end
```

**Comment by gangadhar01a** [comment]
That worked.
Does documentation has this information?

**Comment by PowerSchill** [comment]
No, but I created a new issue to make sure the documentation gets updated.

**Comment by gangadhar01a** [comment]
Thank you!

I have facing 2 issues, with server install resource. 
1) 
Yum is looking for package "rundeck-0:3.0.8.20181029-1.201810292220.noarch" and the dependencies are not available 

install.rb 
rundeck_repository 'rundeck repo' do
  package_uri  'https://dl.bintray.com/rundeck/rundeck-rpm'
  gpgkey       'http://rundeck.org/keys/BUILD-GPG-KEY-Rundeck.org.key'
  gpgcheck     false
end
rundeck_server_install 'Rundeck' do
  action :install
end


"--> Processing Dependency: rundeck-config for package: rundeck-3.0.8.20181029-1.201810292220.noarch
Package rundeck-config is obsoleted by rundeck, but obsoleting package does not provide for requirements
--> Finished Dependency Resolution
Error: Package: rundeck-3.0.8.20181029-1.201810292220.noarch (rundeck)
           Requires: rundeck-config"


2) Trying to install specific version of rundeck, which is throwing an errorrundeck_repository 'rundeck repo' do
  package_uri  'https://dl.bintray.com/rundeck/rundeck-rpm'
  gpgkey       'http://rundeck.org/keys/BUILD-GPG-KEY-Rundeck.org.key'
  gpgcheck     false
end

rundeck_server_install 'Rundeck' do
  version "3.1.1.20190923-1"
  action :install
end

"      * No candidate version available for rundeck-config
      ================================================================================
      Error executing action `install` on resource 'yum_package[rundeck-config]'
"

**Comment by PowerSchill** [comment]
Apparently Rundeck has deprecated the rundeck-config package with version 3.1.0 so now it presents as an error. I am working on an update to fix this now.

**Comment by gangadhar01a** [comment]
Thank you @PowerSchill . Please let me know when the fix is merged.
